pkg/os/disk/validation.go

```diff
@@ -0,0 +1,55 @@
+package disk
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+
+	"kubevirt.io/client-go/log"
+)
+
+const (
+	DiskSourceFallbackPath = "/disk"
+)
+
+func VerifyQCOW2(diskInfo *DiskInfo) error {
+	if diskInfo.Format != "qcow2" {
+		return fmt.Errorf("expected a disk format of qcow2, but got '%v'", diskInfo.Format)
+	}
+
+	if diskInfo.BackingFile != "" {
+		return fmt.Errorf("expected no backing file, but found %v", diskInfo.BackingFile)
+	}
+	return nil
+}
+
+func VerifyImage(diskInfo *DiskInfo) error {
+	switch diskInfo.Format {
+	case "qcow2":
+		return VerifyQCOW2(diskInfo)
+	case "raw":
+		return nil
+	default:
+		return fmt.Errorf("unsupported image format: %v", diskInfo.Format)
+	}
+}
+
+func GetDiskInfoWithValidation(imagePath string, diskMemoryLimitBytes int64) (*DiskInfo, error) {
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("ulimit -t %d && ulimit -v %d && %v info %v --output json", 10, diskMemoryLimitBytes/1024, QEMUIMGPath, imagePath))
```

Comment:
This means the launcher image needs bash. Let's make it visible to other consumers by listing it in release notes.

Reply:
Done

Reply:
won't `sh -c` do the same?

Reply:
Yes, it probably does, but does it matter which one we use?